WW-3748: Add tab reordering functionality with up/down arrows

leo91000 opened the pull request:
## Summary
- Add orderable property to tabIndex field configuration enabling reordering in the editor
- Implement moveTabUp and moveTabDown methods in wwElement for swapping tab positions
- Enable reordering of both tab headers and content panels while maintaining sync

## Test plan
- [ ] Verify up/down arrows appear in tabs field when orderable is enabled
- [ ] Test moving tabs up and down preserves tab content association
- [ ] Confirm disabled state for first tab (up arrow) and last tab (down arrow)
- [ ] Validate reordering works with different numbers of tabs

Ref: https://weweb.atlassian.net/browse/WW-3748

leo91000 committed:
Add AI documentation for ww-tabs

MarcFabre committed:
Update AI.md

leo91000 committed:
Add tab reordering functionality with up/down arrows

- Add orderable property to tabIndex field configuration
- Implement moveTabUp and moveTabDown methods in wwElement
- Enable reordering of both tab headers and content panels

Ref: https://weweb.atlassian.net/browse/WW-3748

leo91000 committed:
feat(WW-3748): Add custom tab labels with double-click editing

- Add tabLabels property for backward-compatible custom labels
- Implement double-click editing on tab labels in editor
- Update move/remove operations to handle custom labels
- Maintain backward compatibility with existing tabs

leo91000 committed:
feat(WW-3748): Add exposed actions and update documentation

- Add exposed actions with proper args for workflow integration
- Update AI.md documentation to match new format
- Include all tab management actions: add, remove, move, updateLabel

leo91000 committed:
docs(WW-3748): Add tabLabels property to AI documentation

- Add tabLabels property description with type and example
- Document custom tab labels array structure
- Include fallback behavior explanation

leo91000 committed:
feat: Add local context properties to tabs component

- Add wwLayoutItemContext for tab headers and content areas
- Provide tabIndex, tabLabel, isActive, totalTabs, and tabPosition context data
- Update AI.md documentation with context data section
- Enable child components to access tab-specific context information

AurelieV commented on src/wwElement.vue:
```diff
@@ -2,26 +2,50 @@
+                    is-repeat 
+                    :index="index - 1" 
+                    :data="{ 
+                        tabIndex: index - 1, 
```
si c'est nouveau, on ne préfère pas plutôt un local context?

leo91000 replied:
Done, j'ai utilisé useRegisterElementLocalContext à la place

AurelieV replied:
@leo91000 useRegisterElementLocalContext c'est pour avoir un contexte au niveau de l'élément lui même, si c'est pour les enfants, il vaut mieux passer par le composant. Du coup c'est good?

leo91000 committed:
refactor(WW-3748): Replace wwLayoutItemContext with useRegisterElementLocalContext

- Remove wwLayoutItemContext components from tab headers and content
- Add useRegisterElementLocalContext with comprehensive tab data and methods
- Provide navigation methods (goToTab, goToNextTab, goToPreviousTab)
- Include complete markdown documentation for local context API
- Ensure consistent event emission for all tab changes
- Maintain backward compatibility with existing tab functionality

leo91000 committed:
docs(WW-3748): Remove usage examples from local context markdown

- Keep only data and methods documentation
- Remove code examples to maintain cleaner documentation

leo91000 committed:
refactor(WW-3748): Replace useRegisterElementLocalContext with wwLocalContext

- Use wwLocalContext with 'tab' key for global context (currentTabIndex, totalTabs, tabPosition, tabLabels)
- Use wwLocalContext with 'tabItem' key for individual tab context (tabIndex, tabLabel, isActive)
- Remove useRegisterElementLocalContext approach for simpler context management
- Provide cleaner separation between global tab state and individual tab item data

leo91000 committed:
fix(WW-3748): Move wwLocalContext inside root div element

- Move global wwLocalContext from template root to inside main div
- Ensure proper DOM structure with div as root element
- Prevent potential issues with context components at template root

leo91000 committed:
refactor(WW-3748): Simplify local context structure to use single tab context

- Replace localKey with elementKey prop for wwLocalContext components
- Remove global tab context wrapper, use individual tab contexts only
- Merge tab data (totalTabs, tabPosition) into each tab context
- Update AI.md documentation to reflect correct context access pattern

leo91000 committed:
fix(WW-3748): Create missing default labels when reordering tabs

leo91000 committed:
fix(WW-3748): Initialize tabLabels with correct numbering when adding tabs
